README.md

```diff
@@ -1,85 +1,135 @@
+combining usage of the `obs_scm` and `tar` source services.
 
-### obs_scm
+### snapcraft
 
-`obs_scm` uses the new `obscpio` archive format to store the sources.
-It is recommended to use `obs_scm` in favour to `tar_scm`, because it
-helps to save a lot of disk space on the server side, especially when
-used in continuous integration (e.g. nightly builds etc.).  The usual
-source tarballs can be regenerated from this at build-time using the
-`tar` and `recompress` source services, so no changes to `.spec` files
-are required when switching to `obs_scm` and `obscpio`.
+The `snapcraft` source service can be used to fetch sources before
+building a [`snappy` app (a.k.a. *snap*)](https://snapcraft.io/).
+
+It parses [a `snapcraft.yaml`
+file](https://docs.snapcraft.io/build-snaps/syntax), looking for any
+[parts in the `parts`
+section](https://docs.snapcraft.io/build-snaps/parts) which have
+[`source-type`](https://docs.snapcraft.io/reference/plugins/source)
+set to one of the supported SCMs.  For each one it will fetch the
+sources via the SCM from the upstream repository, and build a tarball
+from it.
+
+Finally it will write a new version of `snapcraft.yaml` which has the
+`source` value rewritten from the original URL, to the name of the
+part, which is also the name of the newly created local file.  This
+allows the snap to be built purely from local files.
 
-`obs_scm` additionally generates a file named `<package>.obsinfo`
-which includes useful information from your SCM system, so this data
-can be accessed at build-time.
+### appimage
 
-### tar
+The `appimage` source service can be used to fetch sources before
+building an [AppImage app](http://appimage.org/).  It parses [an
+`appimage.yml`
+file](https://github.com/AppImage/AppImages/blob/master/YML.md), looks
+for an optional `build` section at the top-level, and for any sub-key
+named after a supported SCM, it will treat the corresponding value as
+a URL, fetch the sources via the SCM from the upstream repository, and
+build a tarball from it.  You can find example `appimage.yml` files
+under the `tests/fixtures/` subdirectory.
 
-The `tar` source service creates a tarball out of `obscpio` archives.
+## Archive Formats
 
-### snapcraft
+### tar
 
-The `snapcraft` source service can be used to fetch sources before
-building a [`snappy` app (a.k.a. *snap*)](https://snapcraft.io/).  It
-parses a `snapcraft.yml` file, for the SCM related parts and executes
-the corresponding services.
+The standard `tar` archive format is used as output format by the
+`tar` and `tar_scm` source services.
 
-### appimage
+### obscpio
 
-The `appimage` source service can be used to fetch sources before
-building an [AppImage app](http://appimage.org/).  It parses an
-`appimage.yml` file for the SCM related parts and executes the
-corresponding services.
+The `obscpio` archive format is the new OBS specific archive format
+based on `cpio` which can be consumed and generated by the [OBS Delta
```

Comment:
well, it is just a cpio in "newc" format, not really an OBS invention. But we expect .obscpio to be in "newc" format for being able to process to store it incremental indeed.

Reply:
ah, just read about the newc below ;) I wouldn't call it "OBS specific archive format" nevertheless ... Just write

The `obscpio` archives are cpio archives in "newc" format which can be stored incremental on the server side thanks to OBS Delta Store

Reply:
Yep good idea thanks - rewording.